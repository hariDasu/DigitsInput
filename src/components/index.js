import {CodeInput} from "./CodeInput"

import {ProgressBar} from './ProgressBar';

export {CodeInput,  ProgressBar};
