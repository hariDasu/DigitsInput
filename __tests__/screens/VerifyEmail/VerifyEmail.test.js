import React from 'react';
import {render, fireEvent } from '@testing-library/react-native';
import {VerifyEmail} from '../../../src/screens';

describe('Verify email screen testing', () => {
  it('VerifyEmail screen should match the snapshot', () => {
    const {toJSON} = render(<VerifyEmail />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('bottom next button still visible while editing text', () => {
    const {getAllByA11yLabel, getByText} = render(<VerifyEmail />);
    const codeInput = getAllByA11yLabel('verifyCodeInputs');
    fireEvent.changeText(codeInput[0], '1');
    fireEvent.changeText(codeInput[1], '2');

    const nextButton = getByText('Next');
    expect(nextButton).toBeTruthy();
  });
});
