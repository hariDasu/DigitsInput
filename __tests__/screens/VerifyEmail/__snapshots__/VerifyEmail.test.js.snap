// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify email screen testing VerifyEmail screen should match the snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#F6F1EB",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "position": "absolute",
        "top": 60,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessibilityHint="Back Icon"
        accessibilityLabel="backIcon"
        accessible={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <View
        animated={true}
        animationConfig={
          Object {
            "bounciness": 0,
          }
        }
        animationType="spring"
        indeterminate={false}
        indeterminateAnimationDuration={1000}
        onLayout={[Function]}
        progress={0.25}
        style={
          Array [
            Object {
              "backgroundColor": "lightgrey",
              "borderColor": "#308C6A",
              "borderRadius": 0,
              "borderWidth": 0,
              "overflow": "hidden",
              "width": 600,
            },
            undefined,
          ]
        }
        useNativeDriver={false}
      >
        <View
          style={
            Object {
              "backgroundColor": "#308C6A",
              "height": 4,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateX": -225,
                },
                Object {
                  "scaleX": 0.250075,
                },
              ],
            }
          }
        />
      </View>
      <Text
        accessibilityHint="Help Icon"
        accessibilityLabel="helpIcon"
        accessible={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "column",
          "paddingLeft": 20,
          "paddingTop": 10,
          "width": 450,
        }
      }
    >
      <Text
        accessibilityHint="Verify Email text"
        accessibilityLabel="verifyEmailText"
        accessible={true}
        style={
          Object {
            "fontFamily": "Helvetica",
            "fontSize": 24,
            "fontWeight": "bold",
          }
        }
      >
        Verify Email
      </Text>
      <Text
        accessibilityHint="Verify Email Sub text"
        accessibilityLabel="verifyEmailSubText"
        accessible={true}
        style={
          Object {
            "fontFamily": "Helvetica",
            "fontSize": 16,
            "fontWeight": "normal",
            "paddingLeft": 10,
            "textAlign": "left",
          }
        }
      >
        Please enter the code sent to robby@internet.com.
         
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 44,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
          Object {
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 20,
            "width": 280,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
           - 
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 24,
              "textAlign": "center",
            }
          }
        >
           - 
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View
          accessibilityHint="Code verification input"
          accessibilityLabel="verifyCodeInputs"
          accessible={true}
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "height": 60,
                "justifyContent": "center",
                "width": 20,
              },
              false,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 24,
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="characters"
        autoCorrect={false}
        blurOnSubmit={false}
        caretHidden={true}
        clearButtonMode="never"
        disableFullscreenUI={true}
        keyboardType="number-pad"
        maxLength={10}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        onPress={[Function]}
        rejectResponderTermination={true}
        spellCheck={false}
        style={
          Object {
            "bottom": 0,
            "fontSize": 1,
            "left": 0,
            "opacity": 0.01,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        textContentType="oneTimeCode"
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
  </View>
  <View>
    <View
      style={
        Object {
          "paddingBottom": 15,
          "paddingLeft": 20,
        }
      }
    >
      <Text
        accessibilityHint="Didn't receive code text"
        accessibilityLabel="unreceivedCodeText"
        accessible={true}
        style={
          Object {
            "color": "lightgrey",
          }
        }
      >
         
        Didn't receive your code?
         
        <Text
          accessibilityHint="Resend Email Link text"
          accessibilityLabel="resendEmailLinkText"
          accessible={true}
          style={
            Object {
              "color": "#308C6A",
            }
          }
        >
          Resend
        </Text>
      </Text>
      <Text
        accessibilityHint="Update Email description text"
        accessibilityLabel="incorrectEmailText"
        accessible={true}
        style={
          Object {
            "color": "lightgrey",
          }
        }
      >
         
        Enter your email incorrectly?
         
        <Text
          accessibilityHint="Update Email Link text"
          accessibilityLabel="updateEmailText"
          accessible={true}
          style={
            Object {
              "color": "#308C6A",
            }
          }
        >
          Update email
        </Text>
      </Text>
    </View>
    <View
      accessible={true}
      style={
        Object {
          "backgroundColor": "#308C6A",
          "borderRadius": 25,
          "marginBottom": 44,
          "marginLeft": 75,
          "width": 600,
        }
      }
    >
      <View
        accessibilityLabel="verifyEmailNextButton"
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
